types/address_bench_test.go

```diff
@@ -11,6 +11,18 @@ import (
 	"github.com/cosmos/cosmos-sdk/types"
 )
 
+func BenchmarkAccAddressString(b *testing.B) {
+	b.ReportAllocs()
+	pkBz := make([]byte, ed25519.PubKeySize)
+	pk := &ed25519.PubKey{Key: pkBz}
+	aa := pk.Address()
+	var str string
+	for i := 0; i < b.N; i++ {
+		str = aa.String()
+	}
+	require.NotEmpty(b, str)
```

Comment:
To better get a good benchmark in which code won't be optimized away, use a global variable that's an interface e.g.
```go
var sink, revert interface{}
...
for i := 0; i < b.N; i++ {
   result := ...
   sink = result
}
if  sink == nil {
   b.Fatal("benchmark never ran")
}
sink = revert
```

Reply:
You mean a variable outside of any function? We are not using it in other benchmarks.

Reply:
Well just because it wasn't done correctly before doesn't mean we shouldn't do it correctly, I certainly have submitted many other benchmarks where we set to a global sink and check that.

Reply:
I checked it locally - it doesn't change anything. Are you sure we need to trick the benchmark code? 
My only objection is that we make the whole thing less readable and more complex without any observable effect.

Reply:
I am sure about code at times getting optimized away and that being the only way to block that. Humbly, for a bit of clarity, I work on the Go project, across almost all the packages and I am bringing this advice from the many years of working on benchmarking related code :-)

Reply:
I see. But why to obfuscate more if there is no difference (I verified it)? The `require.NoEmpty` already assures that value must not be ignored.